src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java

```diff
@@ -0,0 +1,152 @@
+
+public class DefaultCharsetTest
+{
+    @Test
+    public void testDefaultIfNoPropertyDefined()
+    {
+        Charset systemDefault = Charset.defaultCharset();
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean noCharsetPropertyDefined = definedCharsetName == null;
+        Assume.assumeTrue(noCharsetPropertyDefined);
+
+        assertEquals(systemDefault, clientDefault);
+    }
+
+    @Test
+    public void testDefinedCharset()
+    {
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean charsetPropertyDefined = definedCharsetName != null && !definedCharsetName.isEmpty();
+
+        Assume.assumeTrue(charsetPropertyDefined);
+
+        Charset definedCharset = Charset.forName(definedCharsetName);
+        assertEquals(definedCharset, clientDefault);
+    }
+
+    @Test
+    public void testStaticInitializerValidDefinedCharset()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException
+    {
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+        Charset systemDefault = Charset.defaultCharset();
+        String goodCharsetName = "UTF-16";
+
+        try
+        {
+            Charset goodCharset = Charset.forName(goodCharsetName);
+
+            // Open up the private static initializer
+            Method staticInitializer = DefaultCharset.class.getDeclaredMethod("initializeDefaultCharsetSingleton",
+                                                                              null);
+            staticInitializer.setAccessible(true);
+
+            // Set system property for client charset to bad value
+            System.setProperty(Constants.CLIENT_OPTION_CHARSET, goodCharsetName);
+
+            // Run static initializer to test good charset handling
+            DefaultCharset initializedDefaultCharset = (DefaultCharset) staticInitializer.invoke(null);
+
+            // Get initialized charset
+            Field currentCharsetField = DefaultCharset.class.getDeclaredField("currentCharset");
+            currentCharsetField.setAccessible(true);
+            AtomicReference<Charset> initializedCharset =
+                    (AtomicReference<Charset>)currentCharsetField.get(initializedDefaultCharset);
+
+            assertEquals(goodCharset.name(), initializedCharset.get().name());
+        }
+        finally
+        {
+            // Cleanup
+            if (definedCharsetName != null)
+            {
+                System.setProperty(Constants.CLIENT_OPTION_CHARSET, definedCharsetName);
+            }
+            else
+            {
+                System.clearProperty(Constants.CLIENT_OPTION_CHARSET);
+            }
+        }
+    }
+
+    @Test
+    public void testStaticInitializerInvalidDefinedCharset()
```

Comment:
It might be a good idea to use kind of a [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development) name for the test cases which are intended to freeze a  behavior.  Such approach will shed a bit more light on the expected behavior. For this case it might be something like:

    testTheDefaultSytemCharsetShouldBeUsedIfTheWrongOneWasRequested()

or something like this.